homeassistant/components/songpal/config_flow.py

```diff
@@ -0,0 +1,158 @@
+"""Config flow to configure songpal component."""
+import logging
+from typing import Optional
+from urllib.parse import urlparse
+
+from songpal import Device
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import ssdp
+from homeassistant.const import CONF_HOST, CONF_NAME
+
+from .const import CONF_ENDPOINT, CONF_MODEL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+DEFAULT_NAME = "Songpal"
+
+
+class SongpalConfig:
+    """Device Configuration."""
+
+    def __init__(self, name, host, endpoint, model=None):
+        """Initialize Configuration."""
+        self.name = name
+        self.host = host
+        self.endpoint = endpoint
+        self.model = model
+
+
+class SongpalConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Songpal configuration flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize the flow."""
+        self.conf: Optional[SongpalConfig] = None
+
+    async def _show_setup_form(self, user_input=None, errors=None):
+        user_input = user_input or {}
+        default_name = user_input.get(CONF_NAME) or DEFAULT_NAME
```

Comment:
Is there a simple way to get the name from the device without using ssdp? I'm not very familiar with the API.

Reply:
As soon as there is a Device object, this can be used to obtain the name: https://github.com/rytilahti/python-songpal/blob/master/songpal/device.py#L228 which returns this https://github.com/rytilahti/python-songpal/blob/87ddd69979777d94a14cf7bede0733656487f923/songpal/containers.py#L204.

But my question is, is there a need to even have a name field during the flow at all, or can that be leftto be done after the initialization?